DMCompiler/DM/Expressions/Procs.cs

```diff
@@ -0,0 +1,99 @@
+using System;
+
+namespace DMCompiler.DM.Expressions {
+    // x() (only the identifier)
+    class Proc : DMExpression {
+        string _identifier;
+
+        public Proc(string identifier) {
+            _identifier = identifier;
+        }
+
+        public override void EmitPushValue(DMObject dmObject, DMProc proc) {
+            throw new Exception("attempt to use proc as value");
+        }
+
+        public override ProcPushResult EmitPushProc(DMObject dmObject, DMProc proc) {
+            if (!dmObject.HasProc(_identifier)) {
+                throw new Exception($"Type + {dmObject.Path} does not have a proc named `{_identifier}`");
+            }
+
+            proc.GetProc(_identifier);
+            return ProcPushResult.Unconditional;
+        }
+    }
+
+    // .
+    // This is an LValue _and_ a proc
+    class ProcSelf : LValue {
+        public ProcSelf()
+            : base(null)
+        {}
+
+        public override void EmitPushValue(DMObject dmObject, DMProc proc) {
+            proc.PushSelf();
+        }
+
+        public override ProcPushResult EmitPushProc(DMObject dmObject, DMProc proc) {
+            proc.PushSelf();
+            return ProcPushResult.Unconditional;
+        }
+    }
+
+    // ..
```

Comment:
what about `...` and further friends? :^)

Reply:
unironically a method to get the name of the current proc would be great extension in the future